code/headers/qik_2s12v10.hpp

```diff
@@ -0,0 +1,251 @@
+        void brake_m1(const int8_t brake_amount);
+
+        /** @brief
+         * Returns the errors that the qik2s12v10 has detected since this
+         * function was last used. The meaning of each bit can be found here:
+         * https://www.pololu.com/docs/0J29/5.c
+         *
+         * @return Returns the error if any.
+         */
+        uint8_t get_error_byte();
+
+        /**
+         * @brief
+         * prints the errors returned by the getError function. 
+         * Making debugging easier.
+         * 
+         */
+        void print_errors();
+
+        /** @brief
+         * Returns the value that the parameter specified by parameter
+         * currently has.
+         * @param parameter Specifies the parameter that should be returned.
+         * The parameters can be found here:
+         * https://www.pololu.com/docs/0J29/5.a
+         *
+         * @return Returns the value of the requested parameter.
+         */
+        uint8_t get_configuration_parameter(
+            qik_2s12v10_configuration_parameter parameter);
+
+        /**
+         * @brief
+         * Sets the configuration parameter to the given value and returns if
+         * the parameter was set or if there was an error while setting the
+         * parameter
+         * @param parameter specifies the parameter that should be changed.
+         * @param value specifies what the parameter should be set to in the
+         * chip The parameters can be found here:
+         * https://www.pololu.com/docs/0J29/5.a
+         *
+         * @return returns a parameter_return wich is linked to an error.
+         * eror codes can be found here:
+         * https://www.pololu.com/docs/0J29/5.d
+         */
+        qik_2s12v10_set_configuration_parameter_return
+        set_configuration_parameter(
+            qik_2s12v10_configuration_parameter parameter, uint8_t value);
+
+        /** @brief
+         * Returns the raw reading from motor M0 that indicates how much
+         * current flows through the motor at average over the last 5ms.
+         * This reading is raw, so not converted to milliampere.
+         *
+         * @return Returns the raw current value of the 0 motor.
+         */
+        uint8_t get_m0_current();
+
+        /** @brief
+         * Returns the raw reading from motor M1 that indicates how much
+         * current flows through the motor at average over the last 5ms. This
+         * reading is raw, so not converted to milliampere.
+         *
+         * @return Returns the raw current value of the 1 motor.
+         */
+        uint8_t get_m1_current();
+
+        /** @brief
+         * This function uses the raw reading from motor M0 to estimate how
+         * much current flows through the motor in milliampere. Note that the
+         * value returned by this function can differ from the actual current
+         * by as much as 20% and according to the datasheet the values needs to
+         * be multiplied by 150.
```

Comment:
the multiplication happens internally. This can be a code comment since putting it in doxygen will lead the readers to believe that they still need to do the multiplication themselves.